src/Agent.Worker/Build/TrackingManager.cs

```diff
@@ -228,25 +228,57 @@ public void DisposeCollectedGarbage(IExecutionContext executionContext)
             }
 
             Trace.Info($"Find {gcTrackingFiles.Count()} GC tracking files.");
-            foreach (string gcFile in gcTrackingFiles)
+
+            if (gcTrackingFiles.Count() > 0)
             {
-                try
+                PrintOutDiskUsage(executionContext);
+                foreach (string gcFile in gcTrackingFiles)
                 {
-                    var gcConfig = LoadIfExists(executionContext, gcFile) as TrackingConfig;
-                    ArgUtil.NotNull(gcConfig, nameof(TrackingConfig));
+                    try
+                    {
+                        var gcConfig = LoadIfExists(executionContext, gcFile) as TrackingConfig;
+                        ArgUtil.NotNull(gcConfig, nameof(TrackingConfig));
 
-                    string fullPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), gcConfig.BuildDirectory);
-                    executionContext.Output(StringUtil.Loc("Deleting", fullPath));
-                    IOUtil.DeleteDirectory(fullPath, CancellationToken.None);
+                        string fullPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), gcConfig.BuildDirectory);
+                        executionContext.Output(StringUtil.Loc("Deleting", fullPath));
+                        IOUtil.DeleteDirectory(fullPath, executionContext.CancellationToken);
```

Comment:
Ideally we would also tf delete to cleanup AppData, C:\ProgramData, and server registration. We should add a story so it's at least a known limitation.